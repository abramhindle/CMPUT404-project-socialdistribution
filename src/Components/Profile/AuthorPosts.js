import React, { useEffect, useLayoutEffect, useState } from "react";
import { Panel, PanelGroup } from "rsuite";
import COMMENTS from "../Post/Comment";
import { reqInstance } from "../utils/axios";
import { useParams } from "react-router-dom";
import POST from "../Post/Post";
import EditIcon from "@rsuite/icons/Edit";
import TrashIcon from "@rsuite/icons/Trash";
import { getAuthorId } from "../utils/auth";
import { useNavigate } from "react-router-dom";

function AUTHORPOSTS() {
	const [posts, setPosts] = useState([]);
	let navigate = useNavigate();

	useEffect(() => {
		if (!localStorage.getItem("loggedIn")) {
			navigate("/login");
		} else {
			const author_id = getAuthorId(null);
			const url = `posts/authors/${author_id}/posts/`;
			reqInstance({ method: "get", url: url })
				.then((res) => {
					setPosts(res.data.results);
				})
				.catch((err) => console.log(err));
		}
	}, []);

	const body = (obj) => {
		if (obj["contentType"] === "text/plain") {
			return <p style={{ padding: "5px" }}>{obj["content"]}</p>;
		}

		// Peter you just need to return the image here
		if (obj["contentType"] === "image/jpeg") {
			return <div>image</div>;
		}
	};

	const item = (obj) => {
		return (
			<Panel
				key={obj.id}
				header={<div>{obj["title"]}</div>}
				style={{
					marginTop: "5px",
				}}
				bordered
				collapsible
			>
				<POST postobj={obj} edit={true}></POST>
			</Panel>
		);
	};

	return <PanelGroup>{posts.map((obj) => item(obj))}</PanelGroup>;
}

export default AUTHORPOSTS;
