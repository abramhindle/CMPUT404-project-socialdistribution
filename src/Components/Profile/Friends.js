import React, { useLayoutEffect, useState } from "react";
import TrashIcon from "@rsuite/icons/Trash";
import { Avatar, IconButton, Message, useToaster } from "rsuite";
import { reqInstance } from "../utils/axios";
import { getAuthorId } from "../utils/auth";
import { useNavigate } from "react-router-dom";

function FRIENDS() {
	const [friends, setFriends] = useState({ items: [] });
	let navigate = useNavigate();
	const toaster = useToaster();

	useLayoutEffect(() => {
		if (!localStorage.getItem("loggedIn")) {
			navigate("/login");
		} else {
			const AUTHOR_ID = getAuthorId(null);
			const url = `authors/${AUTHOR_ID}/followers/`;
			reqInstance({
				method: "get",
				url: url,
			}).then((res) => {
				setFriends(res.data);
			});
		}
	}, []);

	async function handleDeleteFollower(obj) {
		const author_id = getAuthorId(null);
		const follower_id = getAuthorId(obj.id);
		const url = `authors/${author_id}/followers/${follower_id}/`;
		reqInstance({ method: "delete", url: url })
			.then((res) => {
				if (res.status === 204) {
					notifySuccessPost();
				} else {
					notifyFailedPost(res.data);
				}
			})
			.catch((err) => console.log(err));
	}
	const notifySuccessPost = () => {
		toaster.push(
			<Message type="success">Successfully removed this follower</Message>,
			{
				placement: "topEnd",
				duration: 5000,
			}
		);
	};
	const notifyFailedPost = (error) => {
		toaster.push(<Message type="error">{error}</Message>, {
			placement: "topEnd",
			duration: 5000,
		});
	};

	const item = (obj) => {
		return (
			<div
				key={obj.id}
				style={{
					height: "50px",
					border: "0.5px solid lightgrey",
					borderRadius: "10px",
					marginTop: "5px",
				}}
			>
				<div style={{ padding: "5px" }}>
					<Avatar
						style={{ float: "left", marginBotton: "5px" }}
						circle
						src={obj["profileImage"]} //{follow[actor][profileImage]} replace this with the actors profile image url
					/>
					<h5
						style={{
							marginLeft: "10px",
							float: "left",
						}}
					>
						{obj["displayName"]}
					</h5>
				</div>
				<div>
					<IconButton
						style={{ float: "right", marginRight: "10px" }}
						appearance="subtle"
						onClick={() => handleDeleteFollower(obj)}
						icon={<TrashIcon />}
					/>
				</div>
			</div>
		);
	};

	return <div>{friends.items.map((obj) => item(obj))}</div>;
}

export default FRIENDS;
