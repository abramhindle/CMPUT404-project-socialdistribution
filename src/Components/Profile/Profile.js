import React, { useEffect, useState } from "react";
import {
	ButtonGroup,
	Panel,
	Button,
	Navbar,
	Nav,
	Input,
	InputGroup,
	Message,
	useToaster,
} from "rsuite";
import FRIENDS from "./Friends";
import AUTHORPOSTS from "./AuthorPosts";
import { useNavigate } from "react-router-dom";
import { reqInstance } from "../utils/axios";
import ADD_FRIEND_MODAL from "../Modals/AddFriendModal";
import {
	getAuthorId,
	getCsrfToken,
	getProfileImageUrl,
	setCurrentUser,
} from "../utils/auth";
import PROFILEIMAGE from "./ProfileImage";

function PROFILE() {
	const [posts, setPosts] = React.useState(true);
	const [appearance, setAppearance] = React.useState({
		posts: "primary",
		friends: "ghost",
	});
	const [author, setAuthor] = useState({});
	let navigate = useNavigate();
	const [open, setOpen] = useState(false);
	const [imageurl, setImage] = useState("");
	const [giturl, setGiturl] = useState("");
	const [count, setCount] = useState(0);
	let toaster = useToaster();

	useEffect(() => {
		if (!localStorage.getItem("loggedIn")) {
			navigate("/login");
		} else {
			setImage(getProfileImageUrl);
			setAuthor(localStorage.getItem("user"));
		}
	}, []);

	const handlePostsBtnClick = () => {
		setPosts(true);
		setAppearance({ posts: "primary", friends: "ghost" });
	};

	const handleFriendsBtnClick = () => {
		setPosts(false);
		setAppearance({ posts: "ghost", friends: "primary" });
	};

	const [curPage, setCurPage] = useState("profile");

	const handleInboxClick = () => {
		navigate("/");
	};

	async function handleLogoutClick() {
		reqInstance.post("accounts/logout/").then((res) => {
			if (res.status === 200) {
				navigate("/login");
			}
		});
	}

	// make a get request to get author and every post the author made and comments on the posts
	// make a get request to get all the friends of an author

	const handleOpen = () => {
		setOpen(true);
	};

	const handleModalClose = () => {
		setOpen(false);
	};

	const notifySuccessPost = (message) => {
		toaster.push(<Message type="success">{message}</Message>, {
			placement: "topEnd",
			duration: 5000,
		});
	};

	const notifyFailedPost = (error) => {
		toaster.push(<Message type="error">{error}</Message>, {
			placement: "topEnd",
			duration: 5000,
		});
	};

	async function handleGitClick() {
		const author_id = getAuthorId(null);
		const url = `authors/${author_id}/`;
		reqInstance({ method: "post", url: url, data: { github: giturl } })
			.then((res) => {
				notifySuccessPost("successfully upadated the giturl");
			})
			.catch((err) => notifyFailedPost(err));
	}

	async function handleImageClick() {
		const author_id = getAuthorId(null);
		const url = `authors/${author_id}/`;
		reqInstance({
			method: "post",
			url: url,
			data: { profileImage: imageurl },
		})
			.then((res) => {
				setCurrentUser(res.data).then(setCount(count + 1));
				notifySuccessPost("successfully upadated the profile url");
			})
			.catch((err) => notifyFailedPost(err.data));
	}

	return (
		<div style={{ padding: "10px", width: "60%", margin: "auto" }}>
			<Navbar>
				<Navbar.Brand>Socially Distrubted</Navbar.Brand>
				<Nav pullRight>
					<Nav.Item onClick={handleLogoutClick}>Logout</Nav.Item>
				</Nav>
				<Nav pullRight>
					<Nav.Item onClick={handleInboxClick}>Inbox</Nav.Item>
				</Nav>
				<Nav pullRight>
					<Nav.Item>Profile</Nav.Item>
				</Nav>
				<Nav pullRight>
					<Nav.Item onClick={handleOpen}>Add Friend</Nav.Item>
				</Nav>
			</Navbar>
			<Panel shaded>
				<PROFILEIMAGE size="lg" />
				<h2 style={{ marginLeft: "10px", float: "left" }}>
					{author["displayName"]}
				</h2>

				<InputGroup inside style={{ marginTop: "5px" }}>
					<Input
						placeholder="Profile Image Url"
						value={imageurl}
						onChange={(e) => setImage(e)}
					/>
					<InputGroup.Button onClick={handleImageClick}>
						Save
					</InputGroup.Button>
				</InputGroup>

				<InputGroup inside style={{ marginTop: "5px" }}>
					<Input
						placeholder="Github Url"
						value={giturl}
						onChange={(e) => setGiturl(e)}
					/>
					<InputGroup.Button onClick={handleGitClick}>
						Save
					</InputGroup.Button>
				</InputGroup>

				<ButtonGroup
					justified
					style={{ paddingTop: "10px", marginBottom: "5px" }}
				>
					<Button
						style={{ textAlign: "center" }}
						appearance={appearance["posts"]}
						onClick={handlePostsBtnClick}
					>
						Posts
					</Button>
					<Button
						style={{ textAlign: "center" }}
						appearance={appearance["friends"]}
						onClick={handleFriendsBtnClick}
					>
						Friends
					</Button>
				</ButtonGroup>
				{posts ? <AUTHORPOSTS /> : <FRIENDS />}
			</Panel>
			<ADD_FRIEND_MODAL open={open} handleClose={handleModalClose} />
		</div>
	);
}

export default PROFILE;
