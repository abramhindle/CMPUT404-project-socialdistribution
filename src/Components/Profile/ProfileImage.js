import React from "react";
import { Avatar } from "rsuite";
import { getProfileImageUrl } from "../utils/auth";

function PROFILEIMAGE({ size }) {
	return (
		<Avatar
			style={{ float: "left" }}
			circle
			src={getProfileImageUrl()}
			size={size}
		></Avatar>
	);
}

export default PROFILEIMAGE;
