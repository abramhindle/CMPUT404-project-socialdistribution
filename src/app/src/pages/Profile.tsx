import * as React from "react"
import { Box, Card, IconButton, Avatar, List, CardContent, Button, Typography } from "@mui/material"
import GitHubIcon from "@mui/icons-material/GitHub"
import PersonIcon from "@mui/icons-material/Person"
import NavBar from "../components/NavBar";

export default function Profile(): JSX.Element {
    const author = {
        displayName: 'John Doe',
        github: 'github',
        profileImage: null,
        friends:2,
        followers:10,
        following:5
    }

    // If it's your profle - Edit
    // If you follow them - Unfollow
    // You sent them a request - Request Sent
    // Else - Follow
    const myProfile = false;
    const [isFollowing, setFollowing] = React.useState(true);
    const [sentRequest, setRequestSent] = React.useState(false);

    const handleFollow = () => {
        setRequestSent(true);
    }

    const handleUnfollow = () => {
        setFollowing(false);                     
    }

    if (author !== undefined){
        return (
        <>
        <Box sx={{ height: window.innerHeight,width: window.innerWidth}}>
            <Box style={{ height: '5%' }} sx={{ bgcolor:"#fff"}}>
                <NavBar items={[
                {
                    Text: "",
                    handleClick: () => {
                    console.log(1);
                    },
                },
                ]} />
            </Box>
            <Box style={{ display: 'flex', height: "95%" }} sx={{ bgcolor:"#fff"}}>
                <Box boxShadow={5} display="flex" sx={{
                        flexDirection: 'column',
                        width: '30%',
                        justifyContent: 'center',
                        alignItems: 'center',
                        bgcolor:"#fff",
                    }}>

                        {author.profileImage?(
                            null
                        ):
                        <Avatar sx={{ width: 150, height: 150, m:2}}>
                            <PersonIcon
                                sx={{ width: 100, height: 100 }}
                            />
                        </Avatar>}
                        
                        <Typography variant="h4" align="center">
                            {author.displayName}
                        </Typography>
                        {author.github?(
                            <IconButton
                                onClick={() =>
                                    window.open(
                                        `https://www.github.com/${author.github}`
                                    )
                                }
                            ><GitHubIcon/>
                            </IconButton>):null}
        
                            <Box sx={{
                                    marginBottom:2,
                                }}>
                                <Typography variant="h6" align="center">
                                Friends: {author.friends}
                                </Typography>
                                <Typography variant="h6" align="center">
                                    Followers: {author.followers}
                                </Typography>
                                <Typography variant="h6" align="center">
                                    Following: {author.following}
                                </Typography>
                            </Box>
                            
                        {myProfile?(
                        <Button
                            variant="contained"
                            onClick={() => {
                                alert('Clicked Edit Button');
                            }}
                            >
                            Edit
                        </Button>
                        ):
                        isFollowing?(
                            <Button 
                            variant="contained"
                            onClick={handleUnfollow}
                            >
                            Unfollow
                            </Button>
                        ):
                        sentRequest?(
                            <Button 
                            variant="contained"
                            disabled
                            >
                            Request Sent
                            </Button>
                        ):
                        <Button 
                        variant="contained"
                        onClick={handleFollow}
                        >
                        Follow
                        </Button>
                        }
        
                </Box>

                <Box overflow="auto" display="flex" sx={{
                    flexDirection: 'column',
                    width: '70%',
                    alignItems: 'center',
                    mt:0.5
                }}>
                    <List style={{maxHeight: '100%', overflow: 'auto'}}>
                        {[1, 2, 3,4,5,6,7,8,9,10].map((i) => (
                            <Card 
                                key={i} 
                                variant="outlined" 
                                sx={{
                                m:2
                                }}
                            >
                                <CardContent sx={{
                                    width: 700,
                                    height:80,
                                    justifyContent: 'center',
                                }}>
                                    Hi
                                </CardContent>
                            </Card>
                        ))} 
                    </List>
                </Box>
            </Box>
        </Box>
        </>
    )
    }
    return <Box/>
}