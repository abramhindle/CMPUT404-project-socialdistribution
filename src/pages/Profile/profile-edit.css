.profile-update-form input {
  height: 2.5em;
  width: 30em;
  border-style: solid;
  border-width: 2pt;
  border-radius: 5pt;
}

.profile-update-form div {
  color: black;
  font-size: 0.5em;
}
