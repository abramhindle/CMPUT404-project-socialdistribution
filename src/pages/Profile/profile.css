div.Page {
  display: flex;
  width: 100vw;
  height: 100vh;
}

.myprofile {
  text-align: center;
  font-size: calc(10px + 2vmin);
  color: white;
  flex: 1;
  float: right;
}

.profileHead {
  background-color: #37342F;
  padding: 1em;
  float: top;
}

.navbar a {
  color: black;
  list-style-type: none;
  margin: 0;
  padding: 1;
  padding-inline: 3em;
  margin-inline: 1em;
  display: inline;
}

.followers {
  list-style: none;
  display: flex;
  flex-wrap: wrap;
}

.followers li {
  border: grey;
  background-color: lightgrey;
  padding: 1em;
  margin-inline: 1em;
  margin-bottom: 1em;
  border-style: solid;
  border-width: 0.5pt;
}

.emptyList {
  color: black;
}

.profileContent {
  padding: 0;
}

.myprofile button {
  margin: 1em;
}