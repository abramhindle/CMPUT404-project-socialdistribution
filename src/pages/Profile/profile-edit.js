import { useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import "./profile-edit.css";
import "../pages.css";

import Sidebar from "../../components/Sidebar/sidebar";
import { update_author } from "../../api/author_api";
import { signInUser } from "../../reducer/userSlice";

const ProfileEdit = () => {
  const user = useSelector((state) => state.user);
  const dispatch = useDispatch();
  const [displayName, setDisplayName] = useState(user.displayName);
  const [github, setGithub] = useState(user.github);
  const [profileImage, setProfileImage] = useState(user.profileImage);

  const submitProfile = async (e) => {
    e.preventDefault();
    update_author(user.id, displayName, github, profileImage, successUpdate);
  };

  const successUpdate = (res) => {
    dispatch(signInUser(res));
  };

  return (
    <div className="Page">
      <div>
        <Sidebar />
      </div>
      <div className="myprofile sidebar-offset">
        <h3 id="title">Update Profile</h3>

        <form className="profile-update-form" onSubmit={submitProfile}>
          <div>
            Display Name
            <input
              type="text"
              value={displayName}
              placeholder="Display Name"
              required
              onChange={(e) => setDisplayName(e.target.value)}
            />
          </div>
          <div>
            Profile Image
            <input
              type="url"
              value={profileImage}
              placeholder="Profile Image (Optional)"
              onChange={(e) => setProfileImage(e.target.value)}
            />
          </div>
          <div>
            Github
            <input
              type="url"
              value={github}
              placeholder="Github (Optional)"
              onChange={(e) => setGithub(e.target.value)}
            />
          </div>
          <button
            id="comment-submit"
            //   disabled={comment ? false : true}
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  );
};

export default ProfileEdit;
