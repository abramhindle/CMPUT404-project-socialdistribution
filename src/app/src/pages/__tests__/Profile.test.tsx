import React from 'react';
import { unmountComponentAtNode } from 'react-dom';
import renderer from 'react-test-renderer';

import Profile from '../Profile';

let container: any = null;
//issues with container type and its methods
const testUser = {"id":"","displayName":"","github":" ","profileImage":" "}

beforeEach(() => {
  // setup a DOM element as a render target
  container = document?.createElement('div');
  document?.body?.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container?.remove();
  container = null;
});

describe('Profile suite', () => {
  it('Renders correctly', () => {
    const tree = renderer.create(<Profile currentUser={testUser}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
