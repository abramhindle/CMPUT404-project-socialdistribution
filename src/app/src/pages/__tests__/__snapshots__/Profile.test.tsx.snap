// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile suite Renders correctly 1`] = `
<div
  className="MuiBox-root css-vykuvf"
>
  <div
    className="MuiBox-root css-1k9ek97"
    style={
      Object {
        "height": "5%",
      }
    }
  >
    <div
      className="sc-bdvvtL dVmOGZ"
    >
      <div
        className="sc-gsDKAQ dokASy"
      >
        <button
          className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-79h83g-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          
        </button>
      </div>
    </div>
  </div>
  <div
    className="MuiBox-root css-1k9ek97"
    style={
      Object {
        "display": "flex",
        "height": "95%",
      }
    }
  >
    <div
      className="MuiBox-root css-v691xz"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-ww6gwo-MuiAvatar-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-tjsfp3-MuiSvgIcon-root"
          data-testid="PersonIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
      <h4
        className="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter css-149xwtl-MuiTypography-root"
      >
        John Doe
      </h4>
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="GitHubIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
          />
        </svg>
      </button>
      <div
        className="MuiBox-root css-1qm1lh"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter css-1kgxqm0-MuiTypography-root"
        >
          Friends: 
          2
        </h6>
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter css-1kgxqm0-MuiTypography-root"
        >
          Followers: 
          10
        </h6>
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter css-1kgxqm0-MuiTypography-root"
        >
          Following: 
          5
        </h6>
      </div>
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Unfollow
      </button>
    </div>
    <div
      className="MuiBox-root css-dhal3f"
    >
      <ul
        className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
        style={
          Object {
            "maxHeight": "100%",
            "overflow": "auto",
          }
        }
      >
        <div
          className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-p1pqv6-MuiPaper-root-MuiCard-root"
        >
          <div
            className="MuiCardContent-root css-gt16ef-MuiCardContent-root"
          >
            Hi
          </div>
        </div>
        <div
          className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-p1pqv6-MuiPaper-root-MuiCard-root"
        >
          <div
            className="MuiCardContent-root css-gt16ef-MuiCardContent-root"
          >
            Hi
          </div>
        </div>
        <div
          className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-p1pqv6-MuiPaper-root-MuiCard-root"
        >
          <div
            className="MuiCardContent-root css-gt16ef-MuiCardContent-root"
          >
            Hi
          </div>
        </div>
        <div
          className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-p1pqv6-MuiPaper-root-MuiCard-root"
        >
          <div
            className="MuiCardContent-root css-gt16ef-MuiCardContent-root"
          >
            Hi
          </div>
        </div>
        <div
          className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-p1pqv6-MuiPaper-root-MuiCard-root"
        >
          <div
            className="MuiCardContent-root css-gt16ef-MuiCardContent-root"
          >
            Hi
          </div>
        </div>
        <div
          className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-p1pqv6-MuiPaper-root-MuiCard-root"
        >
          <div
            className="MuiCardContent-root css-gt16ef-MuiCardContent-root"
          >
            Hi
          </div>
        </div>
        <div
          className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-p1pqv6-MuiPaper-root-MuiCard-root"
        >
          <div
            className="MuiCardContent-root css-gt16ef-MuiCardContent-root"
          >
            Hi
          </div>
        </div>
        <div
          className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-p1pqv6-MuiPaper-root-MuiCard-root"
        >
          <div
            className="MuiCardContent-root css-gt16ef-MuiCardContent-root"
          >
            Hi
          </div>
        </div>
        <div
          className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-p1pqv6-MuiPaper-root-MuiCard-root"
        >
          <div
            className="MuiCardContent-root css-gt16ef-MuiCardContent-root"
          >
            Hi
          </div>
        </div>
        <div
          className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-p1pqv6-MuiPaper-root-MuiCard-root"
        >
          <div
            className="MuiCardContent-root css-gt16ef-MuiCardContent-root"
          >
            Hi
          </div>
        </div>
      </ul>
    </div>
  </div>
</div>
`;
